### DISCUSSION TITLE
Session middleware not enabled for websockets

### DISCUSSION THREAD

**Discussion by callmephilip** [encode/starlette#2741] [Q&A]
It looks like session middleware [explicitly refuses](https://github.com/encode/starlette/blob/master/starlette/middleware/sessions.py#L39) to handle websockets. Is this the reason for  session data not being set in websockets handlers? And if so, why not allow access to session data there?

**Comment by Kludex** [comment]
That line is not refusing to handle websockets... 🤔

**Reply by callmephilip** [reply]
my bad. this was the depth of my research into this. 😆 so session middleware should work with websockets right?

**Reply by callmephilip** [reply]
ok. after some more searching, I can confirm that Websocket.session points to the right session object! works as expected. thank you
